src/Helpers/JWKFetcher.php

```diff
@@ -59,101 +62,135 @@ protected function convertCertToPem($cert)
+    public function getJwksX5c($jwks_url, $kid = null)
     {
-        $r_key = null;
-        if (! $kid) {
-            // No kid indicated, get the first entry.
-            $r_key = $jwks['keys'][0];
-        } else {
-            // Iterate through the JWKS for the correct kid.
-            foreach ($jwks['keys'] as $key) {
-                if (isset($key['kid']) && $key['kid'] === $kid) {
-                    $r_key = $key;
-                    break;
-                }
-            }
+        $cache_key = $jwks_url.'|'.(string) $kid;
+
+        $x5c = $this->cache->get($cache_key);
+        if (! is_null($x5c)) {
+            return $x5c;
         }
 
-        // If a key was not found or the property does not exist, return.
-        if (is_null($r_key) || ! isset($r_key[$prop])) {
+        $jwks = $this->getJwks($jwks_url);
+        $jwk  = $this->getJwk($jwks, $kid);
+
+        if (! $this->subArrayHasOneElement($jwk, 'x5c')) {
             return null;
         }
 
-        // If the value is an array, get the first element.
-        return is_array( $r_key[$prop] ) ? $r_key[$prop][0] : $r_key[$prop];
+        $x5c = $this->convertCertToPem($jwk['x5c'][0]);
+        $this->cache->set($cache_key, $x5c);
+        return $x5c;
     }
 
     /**
      * Get a JWKS given a domain and path to call.
      *
-     * @param string $domain Base domain for the JWKS, including scheme.
-     * @param string $path   Path to the JWKS from the $domain.
+     * @param string $jwks_url URL to the JWKS.
      *
      * @return mixed|string
+     *
+     * @throws RequestException If $jwks_url is empty or malformed.
+     * @throws ClientException  If the JWKS cannot be retrieved.
+     *
+     * @codeCoverageIgnore
      */
-    protected function getJwks($domain, $path)
+    protected function getJwks($jwks_url)
     {
         $request = new RequestBuilder([
-            'domain' => $domain,
-            'basePath' => $path,
+            'domain' => $jwks_url,
+            'basePath' => '',
             'method' => 'GET',
             'guzzleOptions' => $this->guzzleOptions
         ]);
         return $request->call();
     }
+
+    /**
+     * Get a JWK from a JWKS using a key ID, if provided.
+     *
+     * @param array       $jwks JWKS to parse.
+     * @param null|string $kid  Key ID to return; returns first JWK if $kid is null or empty.
+     *
+     * @return null|array
+     *
+     * @codeCoverageIgnore
+     */
+    private function getJwk(array $jwks, $kid = null)
+    {
+        if (! $this->subArrayHasOneElement($jwks, 'keys')) {
+            return null;
+        }
+
+        if (! $kid) {
+            return $jwks['keys'][0];
+        }
+
+        foreach ($jwks['keys'] as $key) {
+            if (isset($key['kid']) && $key['kid'] === $kid) {
+                return $key;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Check if an array within an array has at least one element.
```

Comment:
method is called `subArrayHasOneElement`, not HasAtLeastOne ⚡️ . Choose either but be concise